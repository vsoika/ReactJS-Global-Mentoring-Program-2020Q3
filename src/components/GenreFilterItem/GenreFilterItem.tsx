import React from "react";
import { Nav } from "react-bootstrap";

interface IGenreFilterItemProps {
  genre: string;
}

const GenreFilterItem: React.FC<IGenreFilterItemProps> = ({ genre }) => {
  return (
    <Nav.Item>
      <Nav.Link eventKey={genre}>{genre}</Nav.Link>
    </Nav.Item>
  );
};

export default GenreFilterItem;
