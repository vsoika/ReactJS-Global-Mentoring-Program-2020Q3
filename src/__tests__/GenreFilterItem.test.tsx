import React from "react";
import { render, screen } from "@testing-library/react";
import GenreFilterItem from "../components/GenreFilterItem";

describe("Test suits for GenreFilterItem", () => {
  it("renders GenreFilterItem component", () => {
    const genre = "Documentary";
    const { getByRole } = render(<GenreFilterItem genre={genre} />);
    const anchor = getByRole("button", { name: /Documentary/i });

    screen.debug();
    expect(anchor).toBeInTheDocument();
    expect(anchor).toMatchSnapshot();
  });
});
