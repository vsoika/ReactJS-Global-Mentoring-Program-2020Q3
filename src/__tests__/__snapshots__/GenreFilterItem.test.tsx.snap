// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suits for GenreFilterItem renders GenreFilterItem component 1`] = `
<a
  class="nav-link"
  href="#"
  role="button"
>
  Documentary
</a>
`;
